### ISSUE TITLE
Create a GPU CI job

### ISSUE THREAD

**Issue by kunaltyagi** [PointCloudLibrary/pcl#3575]
## Your Environment

* Operating System and version: CI, default cuda version
* Compiler: nvcc
* PCL Version: HEAD

## Context
Current CI fails to take into account GPU compilation.

## Possible Solution
* adding a GPU CI job
* enabling GPU modules in CI using a gpu enabled base image

**Comment by jasjuang** [comment]
This will be awesome to have but Azure pipelines don't seem to provide instances with GPU. I am curious to know how do you plan to achieve this. For my own downstream project, I am still using good old Jenkins + AWS g3 instances and would love to migrate to a more modern solution if there is one.

**Comment by kunaltyagi** [comment]
> don't seem to provide instances with GPU

For using the compilers, we don't need GPU. GPU are needed only for running.

**Comment by jasjuang** [comment]
@kunaltyagi True, but do we not have unit tests on PCL that requires a GPU to run?

**Comment by kunaltyagi** [comment]
That I don't know. Running tests is at a lower priority right now 😅

**Comment by kunaltyagi** [comment]
I've created a PR, but I don't know how the dockerfiles are built/updated/pushed with correct OS. Don't see any pipeline in there

**Comment by taketwo** [comment]
Up to now this has been done by hand. Creating a pipeline to generate images would a space-age future work :D

**Comment by kunaltyagi** [comment]
Azure has a pipeline for that, with sane defaults. It takes in a file and does the rest. To push it to the registry, that'll need secrets. And pipelines can be set on a timed/manual trigger. Possible before space age, but maybe later
